src/test/java/jenkins/branch/PrimaryBranchHealthMetricTest.java

```diff
@@ -98,8 +103,12 @@ public void given__multibranch_with_primary__when__reporting_health__then__non_e
             assertThat("We now have the master branch", master, notNullValue());
             FreeStyleProject stable = prj.getItem("stable");
             assertThat("We now have the stable branch", stable, notNullValue());
-            assertThat(prj.getBuildHealthReports(), not(containsInAnyOrder(master.getBuildHealth())));
-            assertThat(prj.getBuildHealthReports(), containsInAnyOrder(stable.getBuildHealth()));
+            assertThat(prj.getBuildHealthReports(), not(containsInAnyOrder(
+                    master.getBuildHealthReports().stream().map(Matchers::is).collect(Collectors.toList())
+            )));
+            assertThat(prj.getBuildHealthReports(), containsInAnyOrder(
+                    stable.getBuildHealthReports().stream().map(Matchers::is).collect(Collectors.toList())
+            ));
```

Comment:
A `reason` in these two `assertThat`s would really help explain the condition being tested for.

Reply:
The only reason for the ‘stream’,’map’,’collect’ is that the `containsInAnyOrder(List)` expects a list of matchers not a list of instances

Reply:
We want to verify the list has all the primary branches reports and none of the other’s

Reply:
@stephenc  
Please add `reason` strings to the `assertThat` method calls in the code.

Reply:
Personally I think the reason is redundant once you see past the noise of generic type erasure, but I have added reasons anyway